packages/gestalt-charts/src/ChartGraph/LegendIcon.js

```diff
@@ -0,0 +1,128 @@
+// @flow strict-local
+import { type Node } from 'react';
+import { useColorScheme } from 'gestalt';
+import { useChartContext } from './ChartGraphContext.js';
+import { GraphPoint } from './renderGraphPoint.js';
+import { type DataVisualizationColors } from './types.js';
+
+type Props = {|
+  payloadData:
+    | {|
+        dataKey: string,
+        name: string,
+        stroke: ?string,
+        value: number,
+        strokeDasharray: ?string | number,
+        color: ?string,
+        fill: ?string,
+        legendType?: 'line' | 'rect',
+        isLegend?: boolean,
+      |}
+    | {| referenceArea: 'default', isLegend?: boolean |},
+|};
+
+/**
+ * [LegendIcon](https://gestalt.pinterest.systems/web/chartgraph) component should be used for ....
+ */
+
+function LegendIcon({ payloadData }: Props): Node {
+  const theme = useColorScheme();
+
+  const { decal: showVisualPattern } = useChartContext();
+
+  if (payloadData.referenceArea === 'default') {
+    const dimension = 16;
+
+    return (
+      <svg height={dimension} width={dimension} aria-hidden>
+        <rect
+          style={{ fill: '#ccc', fillOpacity: '0.5' }}
+          height={dimension}
+          width={dimension}
+          rx={2}
+        />
+      </svg>
+    );
+  }
+
+  const { isLegend } = payloadData;
+
+  const colorMap = Object.entries({
```

Comment:
Can we loop through the colorScheme like you did in `usePattern.js` than use a hardcoded map we have to update

Reply:
Not sure if the idea is to get all the colors? or you can use useHex hook you wrote earlier probably.

```
const allColors = useColorScheme()

const allDataVizColors = Object.keys(allColors).filter(key=>key.startsWith('colorDataVisualization')).map(key => allColors[key])
```